engine/src/execution_graph/logic_controllers/BatchWindowFunctionProcessing.cpp

```diff
@@ -214,38 +258,773 @@ ral::execution::task_result ComputeWindowKernel::do_process(std::vector< std::un
+It assumes that preceding_overlaps and following_overlaps will contain N-1 cacheData that corresponds to the preceding and following overlaps copied from the batches
+The idea is that part of batches[x] will be copied to fill preceding_overlaps[x+1] and part will be copied to fill following_overlaps[x-1]
+The preceding_overlaps and following_overlaps will contain metadata to indicate if the overlaps are complete or incomplete (DONE_OVERLAP_STATUS or INCOMPLETE_OVERLAP_STATUS)
+For example, preceding_overlaps[x+1] would be set to INCOMPLETE_OVERLAP_STATUS if batches[x] was not big enough to fulfill the required preceding_value which is defined by the window frame (i.e. ROWS BETWEEN X PRECEDING AND Y FOLLOWING)
+
+The purpose of OverlapAccumulatorKernel is to ensure that all INCOMPLETE_OVERLAP_STATUS overlaps coming from the previous kernel are COMPLETED by copying from other batches.
+The other purpose is to fill the overlaps of preceding_overlaps[0] and following_overlaps[N] with data that has to come from the neighboring nodes (or make a blank overlap if there is no neighbor)
+The OverlapAccumulatorKernel will comunicate with other nodes by sending overlap_requests (PRECEDING_REQUEST, FOLLOWING_REQUEST) which when received and fulfilled are returned as PRECEDING_FULFILLMENT and FOLLOWING_FULFILLMENT
+
+Right before outputting, OverlapAccumulatorKernel will combine preceding_overlaps[x], batches[x] and following_overlaps[x] together to make one batch pushed to the output.
+The following kernel, will then have in one batch with number of rows (preceding_overlaps[x]->num_rows() + batches[x]->num_rows() + following_overlaps[x]->num_rows()), which is the data necessary to procude a batches[x]->num_rows() worth out final output rows.
+
+This kernel uses ConcatenatingCacheDatas a lot to try to reduce and postpone the materialization of data.
+*/
+
+OverlapAccumulatorKernel::OverlapAccumulatorKernel(std::size_t kernel_id, const std::string & queryString,
+    std::shared_ptr<Context> context,
+    std::shared_ptr<ral::cache::graph> query_graph)
+    : distributing_kernel{kernel_id, queryString, context, kernel_type::OverlapAccumulatorKernel} {
+    this->query_graph = query_graph;
+    this->input_.add_port("batches", "preceding_overlaps", "following_overlaps");
+
+    this->num_batches = 0;
+	
+    std::tie(this->preceding_value, this->following_value) = get_bounds_from_window_expression(this->expression);
+
+    ral::cache::cache_settings cache_machine_config;
+	cache_machine_config.type = ral::cache::CacheType::SIMPLE;
+	cache_machine_config.context = context->clone();
+    cache_machine_config.is_array_access = true;
+
+    std::string batches_cache_name = std::to_string(this->get_id()) + "_batches";
+    this->batches_cache = ral::cache::create_cache_machine(cache_machine_config, batches_cache_name);
+    std::string preceding_cache_name = std::to_string(this->get_id()) + "_preceding";
```

Comment:
I will see how i can make that better

Reply:
These caches are internal only to the kernel. They do not connect to other kernels. The names given to these caches here are only so that the logger for the cache machines has a unique name for it. They are not used in the logic anywhere, so I dont think that its necessary to use string constants